azkaban-common/src/main/java/azkaban/utils/PatternLayoutEscaped.java

```diff
@@ -0,0 +1,39 @@
+package azkaban.utils;
+
+import org.apache.log4j.PatternLayout;
+import org.apache.log4j.spi.LoggingEvent;
+
+/**
+ * When we use log4j to send a JSON message the official PatternLayout will send corrupt JSON if certain characters are
+ * present. We use this PatternLayoutEscaped class as a thin interface around PatternLayout to escape all these
+ * characters.
+ */
+public class PatternLayoutEscaped extends PatternLayout {
```

Comment:
I was thinking how a component that uses slf4j would be affected. I guess this change doesn't affect the producer of logging messages. 

wrt log4j2, could you share more details how the migration plan would be? We may eventually move to log4j2.